src/meili/map_matcher.cc

```diff
@@ -261,6 +261,49 @@ MatchResult FindMatchResult(const MapMatcher& mapmatcher,
       return {edge.projected, std::sqrt(edge.distance), next_edge, 0.f, measurement.epoch_time(),
               stateid};
     }
+
+    // when the instersection match fails, we do a more labor intensive search at transitions
+    // for both prev_edge's ending node and next_edge's starting node and see if we could do a
+    // node snap match
+    if (prev_de && edge.id.level() != prev_de->endnode().level()) {
+      baldr::GraphId end_node = prev_de->endnode();
+      tile = graph_reader.GetGraphTile(end_node);
+      for (const auto& trans : tile->GetNodeTransitions(end_node)) {
+        // we only care about if the nodes are in the same level
+        if (trans.endnode().level() != candidate_node.level()) {
+          continue;
+        }
+
+        end_node = trans.endnode();
```

Comment:
we should comment this little block to say that we **can bail** if we cant get the tile at the right level or its the wrong node for the previous edge